pkg/trace/api/profiles.go

```diff
@@ -130,12 +136,21 @@ type multiTransport struct {
 	keys    []string
 }
 
-func (m *multiTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+func (m *multiTransport) RoundTrip(req *http.Request) (rresp *http.Response, rerr error) {
 	setTarget := func(r *http.Request, u *url.URL, apiKey string) {
 		r.Host = u.Host
 		r.URL = u
 		r.Header.Set("DD-API-KEY", apiKey)
 	}
+	defer func() {
+		// Hack for backwards-compatibility
+		// The old v1/input endpoint responded with 200 and as this handler
+		// is just a proxy to existing clients, some clients break on
+		// encountering a 202 response when proxying for the new api/v2/profile endpoints.
+		if rresp != nil && rresp.StatusCode == http.StatusAccepted {
+			rresp.StatusCode = http.StatusOK
```

Comment:
```suggestion
			rresp.Status = http.StatusText(http.StatusOK)
			rresp.StatusCode = http.StatusOK
```
This would be more correct. To avoid any surprises.

Reply:
addressed this in latest commit